ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateRoute/NavigateRoute.cpp

```diff
@@ -0,0 +1,271 @@
+
+    m_navigationButtonEnabled = true;
+    emit navigationButtonChanged();
+  });
+
+  connect(m_routeTask, &RouteTask::createDefaultParametersCompleted, this, [this](QUuid, RouteParameters defaultParameters)
+  {
+    // set values for parameters
+    defaultParameters.setReturnStops(true);
+    defaultParameters.setReturnDirections(true);
+    defaultParameters.setReturnRoutes(true);
+    defaultParameters.setOutputSpatialReference(SpatialReference::wgs84());
+
+    Stop stop1(conventionCenterPoint);
+    Stop stop2(memorialPoint);
+    Stop stop3(aerospaceMuseumPoint);
+
+    QList<Stop> stopsList = {stop1, stop2, stop3};
+    defaultParameters.setStops(stopsList);
+
+    m_routeTask->solveRoute(defaultParameters);
+  });
+
+  connect(m_routeTask, &RouteTask::doneLoading, this, [this](Error error)
+  {
+    if (error.isEmpty())
+    {
+      m_routeTask->createDefaultParameters();
+    }
+    else
+    {
+     qDebug() << error.message() << error.additionalMessage();
+    }
+  });
+}
+
+bool NavigateRoute::navigationButtonEnabled() const
+{
+  return m_navigationButtonEnabled;
+}
+
+bool NavigateRoute::recenterButtonEnabled() const
+{
+  return m_recenterButtonEnabled;
+}
+
+void NavigateRoute::startNavigation()
+{
+  // disable the navigation button
+  m_navigationButtonEnabled = false;
+  emit navigationButtonChanged();
+
+  // get the directions for the route
+  m_directions = m_route.directionManeuvers(this);
+
+  // create a route tracker
+  m_routeTracker = new RouteTracker(m_routeResult, 0, this);
+  connectRouteTrackerSignals();
+
+  // enable "recenter" button when location display is moved from nagivation mode
+  connect(m_mapView->locationDisplay(), &LocationDisplay::autoPanModeChanged, this, [this](LocationDisplayAutoPanMode autoPanMode)
+  {
+    m_recenterButtonEnabled = autoPanMode != LocationDisplayAutoPanMode::Navigation;
+    emit recenterButtonChanged();
+  });
+
+  connect(m_mapView->locationDisplay(), &LocationDisplay::locationChanged, this, [this](const Location& location)
+  {
+    Point projectedLocation = GeometryEngine::project(location.position(), SpatialReference::wgs84());
+    QGeoPositionInfo tempLocation(QGeoCoordinate(projectedLocation.y(), projectedLocation.x()), QDateTime::currentDateTime());
+    m_routeTracker->trackLocation(tempLocation);
+  });
+
+  // turn on map view's navigation mode
+  m_mapView->locationDisplay()->setAutoPanMode(LocationDisplayAutoPanMode::Navigation);
+
+  // add a data source for the location display
+  m_simulatedLocationDataSource = new SimulatedLocationDataSource(this);
+  m_simulatedLocationDataSource->setLocationsWithPolyline(m_route.routeGeometry());
+  m_mapView->locationDisplay()->setDataSource(m_simulatedLocationDataSource);
+  m_simulatedLocationDataSource->start();
+}
+
+void NavigateRoute::connectRouteTrackerSignals()
+{
+  connect(m_routeTracker, &RouteTracker::newVoiceGuidance, this, [this](VoiceGuidance* voiceGuidance)
```

Comment:
do we need to manage the memory for the `voiceGuidance` in any way? e.g. clear it up when we leave scope